sourced/ml/core/models/id_splitter.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import string
+from typing import List, Sequence, Tuple
+
+from modelforge import Model, register_model
+import numpy
+from tensorflow import keras
+from tensorflow.keras.preprocessing.sequence import pad_sequences
+
+from sourced.ml.core.algorithms.id_splitter.nn_model import (f1score, precision,
+                                                             recall)
+from sourced.ml.core.models.license import DEFAULT_LICENSE
+
+
+@register_model
+class IdentifierSplitterBiLSTM(Model):
+    """
+    Bi-Directionnal LSTM Model. Splits identifiers without need for a conventional pattern.
+    Reference: https://arxiv.org/abs/1805.11651
+    """
+    NAME = "id_splitter_bilstm"
+    VENDOR = "source{d}"
+    DESCRIPTION = "Weights of the BiLSTM network to split source code identifiers."
+    LICENSE = DEFAULT_LICENSE
+
+    DEFAULT_MAXLEN = 40
+    DEFAULT_PADDING = "post"
+    DEFAULT_MAPPING = {c: i for i, c in enumerate(string.ascii_lowercase, start=1)}
+    DEFAULT_BATCH_SIZE = 4096
+
+    def construct(self, model: "keras.models.Model",
```

Comment:
@vmarkovtsev in the last modelforge release I do not see `construct` as an abstract method. I think it is kind of deprecated approach and this code should be in`__init__` of the model as we have for https://github.com/src-d/style-analyzer/blob/1da666ab60a12da2c6aab27f736ce6be9354d1b7/lookout/style/typos/model.py#L16-L19 
WDYT?

Reply:
`__init__` declares the empty attributes while `construct` applies the set of new ones so that the object becomes consistent again. Here it is fine.

Reply:
ok, then we should add `__init__ ` with empty attributes declaration.

Reply:
there is no __init__ method used in any of the current modelforge models, however this method is defined in the base class `Model`. Shall I overload it ? Do we have to upgrade every different model ?

Reply:
You need to call `super()`, yes